nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -372,10 +371,10 @@ in
       preStart =  mkOption {
         type = types.lines;
         default = ''
-          test -d ${cfg.stateDir}/logs || mkdir -m 750 -p ${cfg.stateDir}/logs  
-          test `stat -c %a ${cfg.stateDir}` = "750" || chmod 750 ${cfg.stateDir}
-          test `stat -c %a ${cfg.stateDir}/logs` = "750" || chmod 750 ${cfg.stateDir}/logs
-          chown -R ${cfg.user}:${cfg.group} ${cfg.stateDir}
+          test -d '${cfg.stateDir}/logs' || mkdir -p -m 750 '${cfg.stateDir}/logs'
+          test `stat -c %a '${cfg.stateDir}'` = "750" || chmod 750 '${cfg.stateDir}'
+          test `stat -c %a '${cfg.stateDir}/logs'` = "750" || chmod 750 '${cfg.stateDir}/logs'
+          chown -R ${cfg.user}:${cfg.group} '${cfg.stateDir}'
```

Comment:
The chown is no longer require (or effective) as the preStart doesn't run as root anymore. Please remove.

Reply:
Before running nginx requires creation folder '${cfg.stateDir}/logs'

Reply:
The mkdir is fine, only the chown is not required.

Just this line: `chown -R ${cfg.user}:${cfg.group} '${cfg.stateDir}'`